File: Arrays/MinMax.java
package Arrays;
import java.util.*;
public class MinMax {

    public static int[] minMax(int[] arr){
        int[] res = new int[2];
        Arrays.sort(arr);
        res[0] = arr[0];
        res[1] = arr[arr.length-1];
        return res;
    }

    public static int[] minMax2(int[] arr){
        int[] res = new int[2];
        int min = Integer.MAX_VALUE, max = Integer.MIN_VALUE;
        for(int no : arr){
            min = Math.min(min, no);
            max = Math.max(max, no);
        }
        res[0] = min;
        res[1] = max;
        return res;
    }

    public static void main(String[] args){
        int[] arr = {4,2,1,7,5,6};
        int[] res = minMax2(arr);
        System.out.println(res[0]+":"+res[1]);

    }
}

File: Arrays/MinMax/MinMax.go
package main

import (
	"fmt"
	"sort"
)

func method1(arr []int) []int {
	sort.Ints(arr)
	res := make([]int, 0)
	res = append(res, arr[0])
	res = append(res, arr[len(arr)-1])
	return res
}

func method2(arr []int) []int {
	res := make([]int, 0)
	min := 99999
	max := -99999
	for i := 0; i < len(arr); i++ {
		if arr[i] < min {
			min = arr[i]
		}
		if arr[i] > max {
			max = arr[i]
		}
	}
	res = append(res, min)
	res = append(res, max)
	return res
}

func main() {
	arr := []int{3, 2, 1, 5, 4}
	fmt.Println(method1(arr))
	fmt.Println(method2(arr))
}
